File: web-frontend/src/data/datasources/AdDataSource.ts
import { Ad } from '../../domain/entities/Ad';
import { ads } from '../local/ads';

export class AdDataSource {
  async getAds(): Promise<Ad[]> {
    // Simulate API call
    await new Promise(resolve => setTimeout(resolve, 100));
    return ads.map(ad => ({
      ...ad,
      isActive: true,
      createdAt: new Date(),
      updatedAt: new Date()
    }));
  }

  async getAdById(id: string): Promise<Ad | null> {
    const ads = await this.getAds();
    return ads.find(ad => ad.id === id) || null;
  }

  async createAd(ad: Omit<Ad, 'id' | 'createdAt' | 'updatedAt'>): Promise<Ad> {
    const newAd: Ad = {
      ...ad,
      id: Date.now().toString(),
      createdAt: new Date(),
      updatedAt: new Date()
    };
    return newAd;
  }

  async updateAd(id: string, ad: Partial<Ad>): Promise<Ad> {
    const existingAd = await this.getAdById(id);
    if (!existingAd) {
      throw new Error('Ad not found');
    }
    return { ...existingAd, ...ad, updatedAt: new Date() };
  }

  async deleteAd(id: string): Promise<void> { // id 파라미터 추가
    console.log(`Deleting ad with id: ${id}`);
  }
}

File: android-sdk/samplesdk/src/main/java/com/example/android_sdk/data/datasources/AdDataSource.kt
package com.example.android_sdk.data.datasources

import com.example.android_sdk.domain.entities.Ad
import com.example.android_sdk.data.network.ApiService
import kotlinx.coroutines.delay

class AdDataSource(private val apiService: ApiService) {
    suspend fun getAds(): List<Ad> {
        // Simulate network delay
        delay(100)
        return apiService.getAds()
    }

    suspend fun getAdById(id: String): Ad? {
        return apiService.getAdById(id)
    }

    suspend fun createAd(ad: Ad): Ad {
        return apiService.createAd(ad)
    }

    suspend fun updateAd(id: String, ad: Ad): Ad {
        return apiService.updateAd(id, ad)
    }

    suspend fun deleteAd(id: String) {
        apiService.deleteAd(id)
    }
}
